search_service/api/filter.py

```diff
@@ -0,0 +1,95 @@
+from http import HTTPStatus
+from typing import Any, Dict, Iterable  # noqa: F401
+
+from flask_restful import Resource, reqparse
+from flasgger import swag_from
+from marshmallow_annotations.ext.attrs import AttrsSchema
+
+from search_service.api.table import TABLE_INDEX
+from search_service.models.table import SearchTableResultSchema
+from search_service.models.dashboard import SearchDashboardResultSchema
+from search_service.proxy import get_proxy_client
+
+
+class BaseFilterAPI(Resource):
+    """
+    Base Filter API for search filtering
+
+    This API should be generic enough to support every search filter use case.
+    """
+
+    def __init__(self, *, schema: AttrsSchema) -> None:
+        self.proxy = get_proxy_client()
+        self.schema = schema
+        self.parser = reqparse.RequestParser(bundle_errors=True)
+
+        self.parser.add_argument('index', required=False, default=TABLE_INDEX, type=str)
+        self.parser.add_argument('page_index', required=False, default=0, type=int)
+        self.parser.add_argument('query_term', required=False, type=str)
+        self.parser.add_argument('search_request', type=dict)
+
+        super(BaseFilterAPI, self).__init__()
+
+    def post(self) -> Iterable[Any]:
+        """
+        Fetch search results based on the page_index, query_term, and
+        search_request dictionary posted in the request JSON.
+        :return: list of table results. List can be empty if query
+        doesn't match any tables
+        """
+        args = self.parser.parse_args(strict=True)
+        page_index = args.get('page_index')  # type: int
+
+        search_request = args.get('search_request')  # type: Dict
+        if search_request is None:
+            msg = 'The search request payload is not available in the request'
+            return {'message': msg}, HTTPStatus.BAD_REQUEST
+
+        query_term = args.get('query_term')  # type: str
+        if ':' in query_term:
```

Comment:
Didn't know this. Why we don't allow this?

Reply:
mostly due to colon is not allowed or needs to escape in doing query string filter.